modules/rostests/apitests/ntdll/NtFreeVirtualMemory.c

```diff
@@ -167,8 +167,48 @@ static void Test_NtFreeVirtualMemory(void)
     ok(Length == 2*PAGE_SIZE, "Length mismatch : 0x%08lx\n", (ULONG)Length);
     ok(Buffer2 == Buffer, "The buffer is not aligned to PAGE_SIZE.\n");
 }
-    
+
+static void Test_NtFreeVirtualMemory_Parameters(void)
+{
+    NTSTATUS Status;
+    ULONG FreeType;
+    int i;
+
+    // 4th parameter: "ULONG FreeType".
+
+    // A type is mandatory.
+    Status = NtFreeVirtualMemory(NULL, NULL, NULL, 0ul);
+    ok(Status == STATUS_INVALID_PARAMETER_4, "NtFreeVirtualMemory returned status : 0x%08lx\n", Status);
+
+    // All but MEM_DECOMMIT and MEM_RELEASE are unsupported.
+    // Each bit one by one.
+    for (i = 0; i < 32; ++i)
+    {
+        FreeType = 1 << i;
+        if (FreeType == MEM_DECOMMIT || FreeType == MEM_RELEASE)
+            continue;
+
+        Status = NtFreeVirtualMemory(NULL, NULL, NULL, FreeType);
+        ok(Status == STATUS_INVALID_PARAMETER_4, "NtFreeVirtualMemory returned status : 0x%08lx\n", Status);
+    }
+    // All bits at once.
+    // Not (fuzzy-)testing all other values.
```

Comment:
Writing (sentences) like (this) is not making them (more) readable.

Reply:
Removed "(fuzzy-)".